packages/cli/src/metadataGeneration/methodGenerator.ts

```diff
@@ -137,21 +139,27 @@ export class MethodGenerator {
     // we need to throw an error there
     this.path = getPath(decorator, this.current.typeChecker);
     this.produces = this.getProduces();
+    this.consumes = this.getConsumes();
   }
 
-  private getProduces(): string | undefined {
-    const producesDecorators = this.getDecoratorsByIdentifier(this.node, 'Produces');
+  private getProduces(): string[] | undefined {
+    const produces = getProduces(this.node, this.current.typeChecker);
+    return produces.length ? produces : undefined;
+  }
+
+  private getConsumes(): string | undefined {
+    const consumesDecorators = this.getDecoratorsByIdentifier(this.node, 'Consumes');
 
-    if (!producesDecorators || !producesDecorators.length) {
+    if (!consumesDecorators || !consumesDecorators.length) {
       return;
     }
-    if (producesDecorators.length > 1) {
-      throw new GenerateMetadataError(`Only one Produces decorator in '${this.getCurrentLocation()}' method, Found: ${producesDecorators.map(d => d.text).join(', ')}`);
+    if (consumesDecorators.length > 1) {
+      throw new GenerateMetadataError(`Only one Consumes decorator in '${this.getCurrentLocation()}' method, Found: ${consumesDecorators.map(d => d.text).join(', ')}`);
```

Comment:
we allow only one `@Body` per method so I guess it's ok to also allow only one `@Consumes` per method